src/test/java/com/meilisearch/integration/SettingsTest.java

```diff
@@ -844,6 +845,51 @@ public void testUpdateDistinctAttributeSettingsUsingNull() throws Exception {
         assertEquals(initialDistinctAttribute, resetDistinctAttribute);
     }
 
+    /** Tests of the pagination setting methods */
+    @Test
+    @DisplayName("Test get pagination settings by uid")
+    public void testGetPaginationSettings() throws Exception {
+        Index index = createIndex("testGetPaginationSettings");
+        Settings initialSettings = index.getSettings();
+        Pagination initialPagination = index.getPaginationSettings();
+
+        assertEquals(initialSettings.getPagination().getMaxTotalHits(), 1000);
+        assertNotNull(initialPagination.getMaxTotalHits());
+    }
+
+    @Test
+    @DisplayName("Test update pagination settings")
+    public void testUpdatePaginationSettings() throws Exception {
+        Index index = createIndex("testUpdatePaginationSettings");
+        Pagination newPagination = new Pagination();
+
+        Integer MaxTotalHitsTypos = 100;
+
+        newPagination.setMaxTotalHits(MaxTotalHitsTypos);
+        index.waitForTask(index.updatePaginationSettings(newPagination).getTaskUid());
+        Pagination updatedPagination = index.getPaginationSettings();
+
+        assertTrue(updatedPagination.getMaxTotalHits() == 100);
+    }
+
+    @Test
+    @DisplayName("Test reset pagination settings")
+    public void testResetPaginationSettings() throws Exception {
+        Index index = createIndex("testResetPaginationSettings");
+
+        Pagination initialPagination = index.getPaginationSettings();
+        Pagination newPagination = new Pagination();
+        index.waitForTask(index.updatePaginationSettings(newPagination).getTaskUid());
```

Comment:
```suggestion
        Pagination newPagination = new Pagination();

        Integer MaxTotalHitsTypos = 100;
        newPagination.setMaxTotalHits(MaxTotalHitsTypos);
        index.waitForTask(index.updatePaginationSettings(newPagination).getTaskUid());
```
Sorry I made a mistake, these 2 lines are missing for the tests to work, otherwise the pagination is never set to 100 but to 0 ...